### PR TITLE
EVG-6517 Support generate.tasks across deploys

### PR DESCRIPTION
**Pull request by bsamek**
Prior to this commit the retries for generate.tasks are in the body of the job
itself. This means that canceling the job's context would cause it to error.
This commit moves the retry logic into amboy by requeueing the job with an
incrementing ID. This ID is stored in the task document to support the poll job.
The poll job must also check for legacy task IDs, since people may restart old
tasks.

I also snuck in extending the polling interval for jobs.

### CODE REVIEW

**Comment by bsamek** [comment]
I suspect this will cause some tests to fail, since it moves data outside of the job. I'm letting them fail in the PR and then will look again.

File: model/task/task.go [hunk 1]
```diff
@@ -155,6 +156,10 @@ type Task struct {
 	GeneratedTasks bool `bson:"generated_tasks,omitempty" json:"generated_tasks,omitempty"`
 	// GeneratedBy, if present, is the ID of the task that generated this task.
 	GeneratedBy string `bson:"generated_by,omitempty" json:"generated_by,omitempty"`
+	// GeneratedJSON is the the configuration information to create new tasks from.
+	GeneratedJSON []json.RawMessage `bson:"generate_json" json:"generate_json"`
```
Comment (tychoish) [inline model/task/task.go]: isn't this a byte slice rather than a string?

File: rest/data/generate.go [hunk 2]
```diff
@@ -51,10 +63,29 @@ func (gc *GenerateConnector) GeneratePoll(ctx context.Context, taskID string, gr
...
 	}
-	jobID := fmt.Sprintf("generate-tasks-%s", taskID)
-	j, exists := q.Get(ctx, jobID)
-	if !exists {
-		return false, nil, errors.Errorf("task %s not in queue", taskID)
+
+	var jobID string
+	var j amboy.Job
+	var exists bool
+	for {
```
Comment (tychoish) [inline rest/data/generate.go]: check context cancellation in this loop?

File: model/task/task.go [hunk 3]
```diff
@@ -590,6 +594,36 @@ func (t *Task) MarkGeneratedTasks() error {
 	)
 }
 
+// SetGeneratedJSON sets JSON data to generate tasks from.
+func (t *Task) SetGeneratedJSON(json []json.RawMessage) error {
+	t.GeneratedJSON = json
+	return UpdateOne(
```
Comment (tychoish) [inline model/task/task.go]: should we not set this if it's already set?

File: model/task/task.go (hunk 1 above)
Comment (bsamek) [reply model/task/task.go]: Yes, but I don't understand your question. The type is a `[]json.RawMessage` here.

File: model/task/task.go (hunk 3 above)
Comment (bsamek) [reply model/task/task.go]: No, we call this function after we've done the work of generating tasks. The only case where that would matter is if there are multiple jobs running simultaneously, which amboy's machinery should (and does) prevent.

File: rest/data/generate.go (hunk 2 above)
Comment (bsamek) [reply rest/data/generate.go]: done

File: model/task/task.go (hunk 1 above)
Comment (tychoish) [reply model/task/task.go]: Should we not just make it a slice of strings so that it's easier to read in the DB?

**Pull request merged**